crates/kas-widgets/src/scroll_label.rs

```diff
@@ -79,6 +79,16 @@ mod ScrollLabel {
     }
 
     impl Tile for Self {
+        fn role(&self, _: &mut dyn RoleCx) -> Role<'_> {
+            // TODO: this is both a ScrollRegion and Text!
+            Role::Text {
+                text: self.text.as_str(),
+                editable: false,
+                edit_pos: self.selection.edit_pos(),
+                sel_pos: self.selection.sel_pos(),
+            }
+        }
```

Comment:
We can't use `Role::ScrollRegion` and `Role::Text` at the same time. We also can't construct a hierarchy of nodes that doesn't correspond to widgets here (`get_child` needs to return a `Tile` at least, if not a widget).

But I think it is acceptable:

1. The fact that `ScrollLabel` (and `ScrollText`, `EditField`) both scroll and have text is an odd design that limits code-reuse, so refactoring these seems reasonable.
2. We provide the API we need for widgets. If we want to allow introspection of complex text/SVG/canvas/... state later, we can add API appropriate for that (and with the appropriate methods; e.g. coordinate-to-text-position translation should be done through existing text methods, not by exporting a list of all glyph positions).